pkg/discovery/dtofactory/pod_entity_dto_builder_test.go

```diff
@@ -272,3 +273,306 @@ func TestGetRegionZoneLabelCommodity(t *testing.T) {
+			UID:       "test-pod-1-UID",
+		},
+		Spec: api.PodSpec{
+			Affinity: &api.Affinity{
+				NodeAffinity: &api.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &api.NodeSelector{
+						NodeSelectorTerms: []api.NodeSelectorTerm{
+							{
+								MatchExpressions: []api.NodeSelectorRequirement{
+									{
+										Key:      "foo",
+										Operator: api.NodeSelectorOpIn,
+										Values:   []string{"bar", "value2"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			NodeSelector: map[string]string{"test": "foo"},
+		},
+	}
+
+	commoditiesBought, err := builder.getPodCommoditiesBought(mockPod, true, mockPodCommoditiesBoughtTypes)
+
+	assert.Nil(t, err)
+	assert.NotEmpty(t, commoditiesBought)
+	assert.Equal(t, 2, len(commoditiesBought))
+
+	for _, commodity := range commoditiesBought {
+		assert.Equal(t, proto.CommodityDTO_LABEL, *commodity.CommodityType)
+	}
+}
+
+func Test_getPodCommoditiesBought_NodeAffinity(t *testing.T) {
+	feature.DefaultMutableFeatureGate.Set("NewAffinityProcessing=true")
+
+	mockPod := &api.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-1",
+			Namespace: "test-namespace",
+			UID:       "test-pod-1-UID",
+		},
+		Spec: api.PodSpec{
+			Affinity: &api.Affinity{
+				NodeAffinity: &api.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &api.NodeSelector{
+						NodeSelectorTerms: []api.NodeSelectorTerm{
+							{
+								MatchExpressions: []api.NodeSelectorRequirement{
+									{
+										Key:      "foo",
+										Operator: api.NodeSelectorOpIn,
+										Values:   []string{"bar", "value2"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	commoditiesBought, err := builder.getPodCommoditiesBought(mockPod, true, mockPodCommoditiesBoughtTypes)
+
+	assert.Nil(t, err)
+	assert.NotEmpty(t, commoditiesBought)
+	assert.Equal(t, 1, len(commoditiesBought))
+	assert.Equal(t, proto.CommodityDTO_LABEL, *commoditiesBought[0].CommodityType)
+}
+
+func Test_getPodCommoditiesBought_NodeSelector(t *testing.T) {
+	feature.DefaultMutableFeatureGate.Set("NewAffinityProcessing=true")
+
+	mockPod := &api.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-1",
+			Namespace: "test-namespace",
+			UID:       "test-pod-1-UID",
+		},
+		Spec: api.PodSpec{
+			NodeSelector: map[string]string{"test": "foo"},
+		},
+	}
+
+	commoditiesBought, err := builder.getPodCommoditiesBought(mockPod, true, mockPodCommoditiesBoughtTypes)
+
+	assert.Nil(t, err)
+	assert.NotEmpty(t, commoditiesBought)
+	assert.Equal(t, 2, len(commoditiesBought))
+
+	for _, commodity := range commoditiesBought {
+		assert.Equal(t, proto.CommodityDTO_LABEL, *commodity.CommodityType)
+	}
+}
+
+func Test_getPodCommoditiesBought_PodAffinityToSpread(t *testing.T) {
```

Comment:
It looks to me all the new cases with the length of the commodity as 1 are going to the same code branch. 
The `getPodCommoditiesBought` function doesn't seem to treat different kinds of affinity`(pod(node) affinity/anti-affinity)` differently.  Maybe I misunderstand something.

Reply:
I was just trying to ensure that the commodities would be generated for each scenario.

Reply:
Got your point.  thanks for the clarification.